src/sidebar/components/test/filter-status-test.js

```diff
@@ -0,0 +1,384 @@
+      assertButton(createComponent(), {
+        text: 'Show all',
+        icon: null,
+        callback: fakeStore.toggleFocusMode,
+      });
+    });
+  });
+
+  context('(State 6): user-focus mode active, filtered by query', () => {
+    let filterState;
+
+    beforeEach(() => {
+      filterState = {
+        ...getFilterState(),
+        focusActive: true,
+        focusConfigured: true,
+        focusDisplayName: 'Ebenezer Studentolog',
+        filterQuery: 'biscuits',
+      };
+      fakeStore.filterState.returns(filterState);
+      fakeThreadUtil.countVisible.returns(1);
+    });
+
+    it('should show a count of annotations by the focused user', () => {
+      assertFilterText(
+        createComponent(),
+        "Showing 1 result for 'biscuits' by Ebenezer Studentolog"
+      );
+    });
+
+    it('should pluralize annotations when needed', () => {
+      fakeThreadUtil.countVisible.returns(3);
+      assertFilterText(
+        createComponent(),
+        "Showing 3 results for 'biscuits' by Ebenezer Studentolog"
+      );
+    });
+
+    it('should show a no results message when user has no annotations', () => {
+      fakeThreadUtil.countVisible.returns(0);
+      assertFilterText(
+        createComponent(),
+        "No results for 'biscuits' by Ebenezer Studentolog"
+      );
+    });
+
+    it('should provide a "Clear search" button', () => {
+      assertButton(createComponent(), {
+        text: 'Clear search',
+        icon: 'cancel',
+        callback: fakeStore.clearSelection,
+      });
+    });
+  });
+
+  context(
+    '(State 7): user-focus mode active, filtered by query, force-expanded threads',
+    () => {
+      let filterState;
+
+      beforeEach(() => {
+        filterState = {
+          ...getFilterState(),
+          focusActive: true,
+          focusConfigured: true,
+          focusDisplayName: 'Ebenezer Studentolog',
+          filterQuery: 'biscuits',
+          forcedVisibleCount: 2,
+        };
+        fakeStore.filterState.returns(filterState);
+        fakeThreadUtil.countVisible.returns(3);
+      });
+
+      it('should show a count of annotations by the focused user', () => {
+        assertFilterText(
+          createComponent(),
+          "Showing 1 result for 'biscuits' by Ebenezer Studentolog (and 2 more)"
+        );
+      });
+
+      it('should provide a "Clear search" button', () => {
+        assertButton(createComponent(), {
+          text: 'Clear search',
+          icon: 'cancel',
+          callback: fakeStore.clearSelection,
+        });
+      });
+    }
+  );
+
+  context('(State 8): user-focus mode active, selected annotations', () => {
+    let filterState;
+
+    beforeEach(() => {
+      filterState = {
+        ...getFilterState(),
+        focusActive: true,
+        focusConfigured: true,
+        focusDisplayName: 'Ebenezer Studentolog',
+        selectedCount: 2,
+      };
+      fakeStore.filterState.returns(filterState);
```

Comment:
It may be helpful to factor out a helper which maps the logical state ("user focus mode active") to the specific setup logic that is needed. This will reduce the amount of work required if changes are made to how eg. focus mode is represented in future.